smartsim/settings/batch_settings.py

```diff
@@ -140,6 +149,16 @@ def env_vars(self) -> StringArgument:
     @env_vars.setter
     def env_vars(self, value: t.Dict[str, str | None]) -> None:
         """Set the environment variables."""
+
+        if not (
+            isinstance(value, t.Mapping)
+            and all(
+                isinstance(key, str) and isinstance(val, str)
+                for key, val in value.items()
+            )
```

Comment:
Typehint claims to take a `dict[str, str | None]`, but this check will error if I do:
```py
batch_settings.env = {"SOME_KEY": None}
```
Can you resolve this discrepancy rq?

Reply:
changed